tools/src/main/scala/scala/scalanative/build/LLVM.scala

```diff
@@ -31,44 +32,56 @@ private[scalanative] object LLVM {
    *  @return
    *    The paths of the `.o` files.
    */
-  def compile(config: Config, paths: Seq[Path]): Seq[Path] = {
+  def compile(config: Config, paths: Seq[Path])(implicit
+      incCompilationContext: IncCompilationContext = null
+  ): Seq[Path] = {
     // generate .o files for all included source files in parallel
     paths.par.map { path =>
       val inpath = path.abs
       val outpath = inpath + oExt
       val isCpp = inpath.endsWith(cppExt)
       val isLl = inpath.endsWith(llExt)
       val objPath = Paths.get(outpath)
+      val packageName = (config.workdir relativize path).toString
+        .replace(File.separator, ".")
+        .split("\\.")
+        .dropRight(1)
+        .mkString(".")
+
       // LL is generated so always rebuild
       if (isLl || !Files.exists(objPath)) {
-        val compiler = if (isCpp) config.clangPP.abs else config.clang.abs
-        val stdflag = {
-          if (isLl) Seq()
-          else if (isCpp) {
-            // C++14 or newer standard is needed to compile code using Windows API
-            // shipped with Windows 10 / Server 2016+ (we do not plan supporting older versions)
-            if (config.targetsWindows) Seq("-std=c++14")
-            else Seq("-std=c++11")
-          } else Seq("-std=gnu11")
-        }
-        val platformFlags = {
-          if (config.targetsWindows) Seq("-g")
-          else Nil
-        }
-        val expectionsHandling =
-          List("-fexceptions", "-fcxx-exceptions", "-funwind-tables")
-        val flags = opt(config) +: "-fvisibility=hidden" +:
-          stdflag ++: platformFlags ++: expectionsHandling ++: config.compileOptions
-        val compilec =
-          Seq(compiler) ++ flto(config) ++ flags ++
-            asan(config) ++ target(config) ++
-            Seq("-c", inpath, "-o", outpath)
+        if (incCompilationContext == null || incCompilationContext.isChanged(
```

Comment:
this condition can be a part of the previous one, which would eliminate additional nesting level.